cpp/arduino/avr/pgmspace.h

```diff
@@ -50,46 +91,67 @@ out.each { |l| puts d(l) }
 
 #define pgm_get_far_address(var) ( (uint_farptr_t) (&(var)) )
 
-#define memchr_P(...) ::memchr(__VA_ARGS__)
-#define memcmp_P(...) ::memcmp(__VA_ARGS__)
-#define memccpy_P(...) ::memccpy(__VA_ARGS__)
-#define memcpy_P(...) ::memcpy(__VA_ARGS__)
-#define memmem_P(...) ::memmem(__VA_ARGS__)
-#define memrchr_P(...) ::memrchr(__VA_ARGS__)
-#define strcat_P(...) ::strcat(__VA_ARGS__)
-#define strchr_P(...) ::strchr(__VA_ARGS__)
-#define strchrnul_P(...) ::strchrnul(__VA_ARGS__)
-#define strcmp_P(...) ::strcmp(__VA_ARGS__)
-#define strcpy_P(...) ::strcpy(__VA_ARGS__)
-#define strcasecmp_P(...) ::strcasecmp(__VA_ARGS__)
-#define strcasestr_P(...) ::strcasestr(__VA_ARGS__)
-#define strcspn_P(...) ::strcspn(__VA_ARGS__)
-#define strlcat_P(...) ::strlcat(__VA_ARGS__)
-#define strlcpy_P(...) ::strlcpy(__VA_ARGS__)
-#define strnlen_P(...) ::strnlen(__VA_ARGS__)
-#define strncmp_P(...) ::strncmp(__VA_ARGS__)
-#define strncasecmp_P(...) ::strncasecmp(__VA_ARGS__)
-#define strncat_P(...) ::strncat(__VA_ARGS__)
-#define strncpy_P(...) ::strncpy(__VA_ARGS__)
-#define strpbrk_P(...) ::strpbrk(__VA_ARGS__)
-#define strrchr_P(...) ::strrchr(__VA_ARGS__)
-#define strsep_P(...) ::strsep(__VA_ARGS__)
-#define strspn_P(...) ::strspn(__VA_ARGS__)
-#define strstr_P(...) ::strstr(__VA_ARGS__)
-#define strtok_P(...) ::strtok(__VA_ARGS__)
-#define strtok_P(...) ::strtok(__VA_ARGS__)
-#define strlen_P(...) ::strlen(__VA_ARGS__)
-#define strnlen_P(...) ::strnlen(__VA_ARGS__)
-#define memcpy_P(...) ::memcpy(__VA_ARGS__)
-#define strcpy_P(...) ::strcpy(__VA_ARGS__)
-#define strncpy_P(...) ::strncpy(__VA_ARGS__)
-#define strcat_P(...) ::strcat(__VA_ARGS__)
-#define strlcat_P(...) ::strlcat(__VA_ARGS__)
-#define strncat_P(...) ::strncat(__VA_ARGS__)
-#define strcmp_P(...) ::strcmp(__VA_ARGS__)
-#define strncmp_P(...) ::strncmp(__VA_ARGS__)
-#define strcasecmp_P(...) ::strcasecmp(__VA_ARGS__)
-#define strncasecmp_P(...) ::strncasecmp(__VA_ARGS__)
-#define strstr_P(...) ::strstr(__VA_ARGS__)
-#define strlcpy_P(...) ::strlcpy(__VA_ARGS__)
-#define memcmp_P(...) ::memcmp(__VA_ARGS__)
+inline const void * memchr_P(const void *s, int val, size_t len) { return memchr(s, val, len); }
```

Comment:
What's the advantage to the inline wrapper vs the macro?

Reply:
That it is only interpreted as a top level function, and works better with namespaces. I.e. consider:

```
void *foo = ::memchr_P(s, val, len);
```
Which is top namespace qualified to prevent conflicts, which would expand to `::::memchr(...)` with the macro and break.

Also, functions are better than macros when methods of the same name are used. Consider:
```
class Foo {
  void * memchr_P(const void *s, int val, size_t len);
};
```

The macro would be replaced here and break compilation.

Reply:
I retract my question... for some reason I thought that `memcpy_P` was defined in Arduino itself as a macro, and that we were going into new territory by making it an actual function.  Just did a quick google search, and it looks like it was a function to begin with -- so I think this is fine.